README.md

```diff
@@ -58,53 +58,61 @@ type Config struct {
-	// use the `dials_env` tag. Now Dials will lookup "configpath" env value to
-	// populate the Path field
+	// Go convention for environment variables and will look for the dials tag
+	// or field name in all caps when struct tags aren't specified. Without any
+	// struct tags, it would lookup the PATH environment variable. To specify a
+	// different env variable, use the `dials_env` tag. Now Dials will lookup
+	// "configpath" env value to populate the Path field.
 	Path string `dials_env:"configpath"`
 }
 
-// ConfigPath returns the path to the config file. This is particularly helpful
-// when the path is populated from environment variables or command line flags.
-// Dials will first read from environment variables and command line flags
-// and then read the config file specified by the populated field.
+// ConfigPath returns the path to the config file that Dials should read. This 
+// is particularly helpful when it's desirable to specify the file's path via
+// environment variables or command line flags. Dials will first populate the 
+// configuration struct from environment variables and command line flags
+// and then read the config file that the ConfigPath() method returns
 func (c *Config) ConfigPath() (string, bool) {
 	// can alternatively return empty string and false if the state of the
-	// struct doesn't specify a config file to read
+	// struct doesn't specify a config file to read. We would recommend to use
+	// dials.Config directly (shown in the next example) instead of the ez
+	// package if you just want to use environment variables and flags without a
+	// file source
 	return c.Path, true
 }
 
 func main() {
 	c := &Config{}
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	// The following function will populate the config struct by reading the
 	// config files, environment variables, and command line flags (order matches
 	// the function name) with increasing precedence. In other words, the flag
 	// source (last) would overwrite the YAML source (first) were they both to
 	// attempt to set the same struct field. There are several options that can be
 	// passed in as well to indicate whether the file will be watched and updates
 	// to the file should update the config struct and if the flags name component
-	// separation should use different encoding
-	d, dialsErr := ez.YAMLConfigEnvFlag(context.Background(), c, WithWatchingConfigFile())
+	// separation should use different encoding than lower-kebab-case
+	d, dialsErr := ez.YAMLConfigEnvFlag(ctx, c, ez.WithWatchingConfigFile(true))
 	if dialsErr != nil {
 		// error handling
 	}
 
-	// get the struct populated from config file, environment variables, and
-	// command line flags.
-	config := &Config{}
-	d.Fill(config)
-	fmt.Printf("Config: %+v\n", config)
+	// Fill will make a deep copy of the struct and populate it with values from
+	// config file, environment variables, and command line flags. Can
```

Comment:
This is a little confusing.
How about: `Fill will deep copy the fully-stacked configuration into its argument. The stacked configuration is populated from the config file, environment variables and commandline flags.`

Also, let's start the next sentence as `One can` rather than `Can`, so there's a subject.